[ Master FAQ  Bug 11158 ]- Only ro priviledge's user cannot search FAQ

s7design opened the pull request:
http://bugs.otrs.org/show_bug.cgi?id=10923

Hi @carlosfrodriguez ,

Hereby is proposed fix for this issue. Before users with 'ro' permission in FAQ group couldn't search for FAQ's, since category search was based on users with 'rw' permission.

Please let me know what do you think about this proposal.

Regards,
Sanjin

s7design commented:
Hi @carlosfrodriguez ,

Please wait with follow-up this PR. I saw that there are something similar in the bug http://bugs.otrs.org/show_bug.cgi?id=11158. We will look into this and then I will inform you about final solution.

If you have any sugestion in this moment feel free to tell us.

Regards
Zoran

carlosfrodriguez commented:
Hello @s7design,

Thank you for your pull request, it apparently solves the issue as mentioned in the bug report. Hoewever
it looks like there is another very similar pull https://github.com/OTRS/FAQ/pull/14 that also change the 'rw' to 'ro' in other places, can you please analyze if its needed also those changes and marge both pull requests? possibly also updating AgentFAQSearchSmall.

((enjoy))
cr

s7design commented:
Hi @carlosfrodriguez 

Thanks for the follow-up. Yes, you are right. I saw this PR. We have already checked this, but we will look into again. I my opinion after testing that I believe that changes in the PR #14  is not necessary. However, please wait that I check this with more test cases.

Regards
Zoran .

s7design commented:
Hello @carlosfrodriguez ,

Updated PR with change that users with 'ro' permission can also search for categories. 

PR from noritnk: https://github.com/OTRS/FAQ/pull/14 additionally allowed for search of users with 'ro' permission to be listed in 'Created by Agent' and 'Last changed by Agent' FAQ searh. I don't think it is necessary, since if user doesn't have write permission then he won't be able to either create or modify existing FAQ's, resulting in no need to list them in possible FAQ search outcome.

AgentFAQSearchSmall is also updated to reflect changes from AgentFAQSearch module.

Please let me know if there are any further questions or issues regarding this PR.

Regards,
Sanjin

s7design commented:
Hi @carlosfrodriguez

I rebased this PR to make it up to date with the changes which have been happened in meantime.

Regards
Zoran

carlosfrodriguez commented:
Hi @s7design,

Your pull request was merged in https://github.com/OTRS/faq/commit/2fc48f5c6c0f2fc5019186d286cb726713ea8fbe.

Thank you!

((enjoy))
cr
